src/main/java/com/hierynomus/sshj/transport/IdentificationStringParser.java

```diff
@@ -17,20 +17,21 @@
 
 import net.schmizz.sshj.common.Buffer;
 import net.schmizz.sshj.common.ByteArrayUtils;
+import net.schmizz.sshj.common.LoggerFactory;
 import net.schmizz.sshj.transport.TransportException;
 import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 import java.util.Arrays;
 
 public class IdentificationStringParser {
-    private static final Logger logger = LoggerFactory.getLogger(IdentificationStringParser.class);
+    private final Logger log;
     private final Buffer.PlainBuffer buffer;
 
     private byte[] EXPECTED_START_BYTES = new byte[] {'S', 'S', 'H', '-'};
 
-    public IdentificationStringParser(Buffer.PlainBuffer buffer) {
+    public IdentificationStringParser(LoggerFactory loggerFactory, Buffer.PlainBuffer buffer) {
+	this.log = loggerFactory.getLogger(IdentificationStringParser.class);
```

Comment:
I fixed the indentation (sorry I use tabs for 8 spaces reflexively).  What’s up with the “license” issue?